### PR TITLE
Add era display option

### LINKED ISSUE #4478: Add eraDisplay support to NeoDateTimeFormatter
Once open design questions are resolved in https://github.com/tc39/proposal-intl-eradisplay, we should implement it in ICU4X NeoDateTimeFormatter. I already have some stub code that needs to be filled in with the final algorithm.

### PR DESCRIPTION
**Pull request by sffc**
#1317

Part of #4478

This should resolve #4478 for the purposes of 2.0, but we likely want to add more options once https://github.com/tc39/proposal-intl-eradisplay/issues/9 is resolved by CLDR.

CC @Louis-Aime

### CODE REVIEW

**Review by Manishearth — APPROVED** [review]
quick skim looks fine. not a thorough review

File: components/datetime/src/raw/neo.rs [hunk 1]
```diff
@@ -174,10 +185,50 @@ impl DatePatternSelectionData {
...
+                        | Some(AnyCalendarKind::Japanese)
+                        | Some(AnyCalendarKind::JapaneseExtended)
+                        | Some(AnyCalendarKind::Persian)
+                        | Some(AnyCalendarKind::Roc) => true,
+                        Some(AnyCalendarKind::Chinese)
+                        | Some(AnyCalendarKind::Dangi)
+                        | Some(AnyCalendarKind::Hebrew) => false,
+                        Some(AnyCalendarKind::Gregorian) => match datetime.year() {
+                            None => true,
+                            Some(year) if year.number < 1000 => true,
```
Comment (robertbastian) [inline components/datetime/src/raw/neo.rs:216]: is this spec-driven or something we came up with?

File: components/datetime/src/neo_skeleton.rs [hunk 2]
```diff
@@ -73,6 +73,33 @@ impl NeoSkeletonLength {
...
+pub enum EraDisplay {
+    /// Always display the era.
+    ///
+    /// Examples:
+    ///
+    /// - `1000 BC`
+    /// - `77 AD`
+    /// - `2024 AD`
+    Always,
+    /// Display the era when needed to disambiguate the year.
```
Comment (robertbastian) [inline components/datetime/src/neo_skeleton.rs:90]: document this better. iiuc it's for years 0..1000, is that guaranteed?

**Review by robertbastian — APPROVED** [review]

File: components/datetime/src/raw/neo.rs (hunk 1 above)
Comment (robertbastian) [reply components/datetime/src/raw/neo.rs:216]: ah found it in the spec

File: components/datetime/src/neo_skeleton.rs (hunk 2 above)
Comment (sffc) [reply components/datetime/src/neo_skeleton.rs:90]: I'll come back and update this spec text once a decision is reached on the era display design in https://docs.google.com/document/d/1dmMk_XODm3DGe84GmMVw7O6a7oIs5yojDXm_25CcMbw/edit

**Pull request merged**